druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/entity/CacheActionRunner.java

```diff
@@ -78,14 +78,16 @@ public String handleDecode(DecodeConfig decodeConfig, Message parsedMessage, Des
         }
     }
 
-    synchronized public void updateCache(ProtobufSchemaFactory protobufSchemaFactory
+    synchronized public void updateCache(BaseSchemaFactory schemaFactory
             , final String key
             , final byte[] value
             , RocksDB db
             , ServiceEmitter serviceEmitter
             , RocksDBExtractionNamespace extractionNamespace) {
         if (extractionNamespace.isCacheEnabled()) {
             try {
+                validateSchemaFactory(schemaFactory);
```

Comment:
it would be better to use bounded type parameter and restrict this class to be used only with `ProtobufSchemaFactory`. with this it will be a compile time check as against current runtime check.

Reply:
schema factory is compile time validation now, had to add extra logic at the rocksdb namespace